daemon/api_prompting.go

```diff
@@ -97,6 +100,228 @@ func getUserID(r *http.Request) (uint32, Response) {
 	return uint32(userIDInt), nil
 }
 
+type invalidReason string
+
+const (
+	unsupportedValueReason invalidReason = "unsupported-value"
+	parseErrorReason       invalidReason = "parse-error"
+	validationErrorReason  invalidReason = "validation-error"
+)
+
+type invalidFieldValue struct {
+	Reason   invalidReason `json:"invalid-reason"`
+	Metadata interface{}   `json:"metadata,omitempty"`
+}
+
+type UnsupportedValueValue struct {
+	err *errortypes.UnsupportedValueError
+}
+
+func (v *UnsupportedValueValue) MarshalJSON() ([]byte, error) {
+	value := make(map[string]invalidFieldValue, 1)
+	value[v.err.Field] = invalidFieldValue{
+		Reason: unsupportedValueReason,
+		Metadata: &struct {
+			Received  interface{} `json:"received-invalid"`
+			Supported []string    `json:"supported"`
+		}{
+			Received:  v.err.Unsupported,
+			Supported: v.err.Supported,
+		},
+	}
+	return json.Marshal(value)
+}
+
+type ParseErrorValue struct {
+	err *errortypes.ParseError
+}
+
+func (v *ParseErrorValue) MarshalJSON() ([]byte, error) {
+	value := make(map[string]invalidFieldValue, 1)
+	value[v.err.Field] = invalidFieldValue{
+		Reason: parseErrorReason,
+		Metadata: &struct {
+			Received string `json:"received-invalid"`
+			// TODO: once documentation exists for user-defined fields
+			// DocumentationURL string `json:"documentation"`
+		}{
+			Received: v.err.Invalid,
+			// TODO: once documentation exists for user-defined fields
+			// DocumentationURL: <url>,
+		},
+	}
+	return json.Marshal(value)
+}
+
+type validationErrorType string
+
+var (
+	requestedPathNotMatched        validationErrorType = "requested-path-not-matched"
+	requestedPermissionsNotMatched validationErrorType = "requested-permissions-not-matched"
+	ruleConflict                   validationErrorType = "conflict-with-existing-rule"
+)
```

Comment:
this don't feel quite feel like invalid field cases

Reply:
I think the first two are: invalid "path-pattern" field because it didn't match the requested path, or invalid "permissions" field because it didn't match the requested permissions. They could just as easily be their own error kinds though, if you'd prefer.

The rule conflict error is a bit tougher though, since it's really that the combination of the "path-pattern" and "permissions" fields results in a conflict.

I think it also comes down to what it means for a field to be "invalid". If we take that to mean "malformed", I agree, these should be something else... Perhaps it would be simplest just for each of these to be a dedicated error kind.